### PR TITLE
fix #184061 split measure must break notes in other Voices using toDurationList()

### PR DESCRIPTION
**Pull request by ericfont**
Splitting a measure via cmdSplitMeasure() may require that notes in other voices break apart their duration into something that cannot be represented with just a single note.  Therefore it is necessary to use toDurationList() to break the note's duration up into multiple durations that are tied together to represent the original note before the split.

### CODE REVIEW

File: libmscore/range.cpp [hunk 1]
```diff
@@ -314,8 +314,7 @@ bool TrackList::write(Measure* measure) const
       Fraction pos;
       Measure* m       = measure;
       Score* score     = m->score();
-      Fraction rest    = m->len();
-      //Staff* staff     = score->staff(track2staff(_track));
+      Fraction remains = m->len();
```
Comment (ericfont) [inline libmscore/range.cpp]: note that I changed this variable name to better represent that it represents remaining time in measure, but didn't change behavior around it.

**Comment by ericfont** [comment]
@lasconic this is ready for review.

File: libmscore/range.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [inline libmscore/range.cpp]: As mentioned elsewhere, "Rest" is German for "remainder" :-)

**Comment by ericfont** [comment]
I'm going to start work on https://musescore.org/en/node/166096 ...i'll probably just add a commit to this PR to fix that if I finish before this gets merged.

File: libmscore/range.cpp [hunk 2]
```diff
@@ -369,45 +368,62 @@ bool TrackList::write(Measure* measure) const
...
-                                    segment->add(c);
-                                    duration -= d;
-                                    rest     -= d;
-                                    pos      += d;
-                                    foreach(Note* note, c->notes()) {
-                                          if (!duration.isZero() || note->tieFor()) {
-                                                Tie* tie = new Tie(score);
-                                                note->add(tie);
+                                    QList<TDuration> durationList = toDurationList(d, true);
+                                    for (int i = 0; i < durationList.size() ; i++) {
```
Comment (ericfont) [inline libmscore/range.cpp]: oh silly me, the variable name "i" is already used for track scope.

**Comment by lasconic** [comment]
I commited the tests in master too 60d97e7d39563cf1e4eb38f547fbcde3b1a8d851

**Comment by ericfont** [comment]
Great, thanks!

**Pull request merged**